src/sources/util/unix_datagram.rs

```diff
@@ -22,15 +25,27 @@ pub fn build_unix_datagram_source<D>(
     max_length: usize,
     host_key: String,
     decoder: D,
+    decoding: Option<DecodingConfig>,
     shutdown: ShutdownSignal,
     out: Pipeline,
-    build_event: impl Fn(&str, Option<Bytes>, &str) -> Option<Event> + Clone + Send + Sync + 'static,
-) -> Source
+    build_event: impl Fn(
+            &str,
+            Option<Bytes>,
+            Bytes,
+            &(dyn Fn(Bytes) -> crate::Result<Value> + Send + Sync),
+        ) -> Option<Event>
+        + Clone
+        + Send
+        + Sync
+        + 'static,
```

Comment:
IMO this is begging for a type alias to make this overall function signature (and that of `listen` below, as well as the corresponding on in `unix_stream.rs`) more legible.

Reply:
I'm not sure if this is possible for `impl` Traits. Could change it if we don't mind one extra indirection

Reply:
You are of course right, sadly. This would tend to argue for a new trait parameter, but that too has visual overhead.